langpuppet/merge.go

```diff
@@ -0,0 +1,167 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package langpuppet implements an integration entrypoint that combines lang and Puppet.
+package langpuppet
+
+import (
+	"fmt"
+	"github.com/purpleidea/mgmt/engine"
+	"github.com/purpleidea/mgmt/pgraph"
+	"strings"
+)
+
+// This method returns the merged graph containing all vertices
+// and edges found in the graphs produced by the lang and Puppet GAPIs
+// associated with the wrapping GAPI.
+// Vertices are merged if they adhere to the following rules (for any
+// given value of POSTFIX):
+//  1. The graph from lang contains a noop vertex named puppet_POSTFIX
+//  2. The graph from Puppet contains an empty class mgmt_POSTFIX
+//  3. The resulting graph will contain one noop vertex named POSTFIX
+//     that replaces all nodes mentioned in (1) and (2)
+// All edges connecting to any of the vertices merged this way
+// will be present in the merged graph.
+func mergeGraphs(graphFromLang, graphFromPuppet *pgraph.Graph) (*pgraph.Graph, error) {
+	lg := graphFromLang
```

Comment:
why the unnecessary copy?

Reply:
It was easier to type the code out =D

Undone.